compiler-plugin/src/test/kotlin/com/google/devtools/ksp/processor/ReplaceWithErrorTypeArgsProcessor.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Google LLC
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.devtools.ksp.processor
+
+import com.google.devtools.ksp.getClassDeclarationByName
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.symbol.*
+
+open class ReplaceWithErrorTypeArgsProcessor : AbstractTestProcessor() {
+    val results = mutableListOf<String>()
+
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        val decls = listOf(
+            // Those have 2 type parameters
+            resolver.getClassDeclarationByName("KS")!!,
+            resolver.getClassDeclarationByName("KL")!!,
+            resolver.getClassDeclarationByName("JS")!!,
+            resolver.getClassDeclarationByName("JL")!!,
+            // Those have 0 or 1 parameters
+            resolver.getClassDeclarationByName("KS1")!!,
+            resolver.getClassDeclarationByName("KL1")!!,
+            resolver.getClassDeclarationByName("JS1")!!,
+            resolver.getClassDeclarationByName("JL1")!!,
+            resolver.getClassDeclarationByName("JSE")!!,
+            resolver.getClassDeclarationByName("JSE.E")!!,
+            resolver.getClassDeclarationByName("JLE")!!,
+            resolver.getClassDeclarationByName("JLE.E")!!,
+            resolver.getClassDeclarationByName("KSE")!!,
+            resolver.getClassDeclarationByName("KSE.E")!!,
+            resolver.getClassDeclarationByName("KLE")!!,
+            resolver.getClassDeclarationByName("KLE.E")!!,
+        )
+        val x = resolver.getPropertyDeclarationByName(resolver.getKSNameFromString("x"), true)!!
+        val xargs = x.type.element!!.typeArguments
+        val y = resolver.getPropertyDeclarationByName(resolver.getKSNameFromString("y"), true)!!
+        val yargs = y.type.element!!.typeArguments
+
+        for (decl in decls) {
+            results.add(decl.asStarProjectedType().replace(xargs).toString())
```

Comment:
what's the difference between calling and not calling `asStarProjectedType()`?

Reply:
might help reading test case if the name of the declaration is put in the beginning of lines?

Reply:
1. Because there are two different APIs :) `KSType.replace` and `KSClassDeclaration.asType`.
2. Done improving the readability of the outputs.